pkg/replicatedapp/api.go

```diff
@@ -67,6 +84,16 @@ func getLicenseFromAPI(url string, licenseID string) (*LicenseData, error) {
 
 	req.SetBasicAuth(licenseID, licenseID)
 
+	l, err := getLicenseById(licenseID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get license by id")
+	}
+
+	if l != nil {
+		reportingInfo := reporting.GetReportingInfoByAppSlug(l.Spec.AppSlug)
```

Comment:
The app slug is generated and won't always match the one in the license.  I think you will need to use `ListInstalledApps` and then `GetLatestLicenseForApp` and then just call `GetReportingInfo` with the right app ID